examples/RetrieveAuditEvents.py

```diff
@@ -0,0 +1,139 @@
+    for i in range(len(rawoutput.data)):
+        listofregions.append(str(rawoutput.data[i].name))
+    return listofregions
+
+
+def paginate(operation, *args, **kwargs):
+    '''Function to paginate API calls.
+
+    Refer to
+    https://github.com/oracle/oci-python-sdk/blob/master/docs/quickstart.rst
+
+    For better exception handling refer to
+    https://oracle-cloud-infrastructure-python-sdk.readthedocs.io/en/latest/api/index.html#oci.exceptions.ServiceError
+    '''
+    while True:
+        try:
+            response = operation(*args, **kwargs)
+            for value in response.data:
+                yield value
+            kwargs["page"] = response.next_page
+            if not response.has_next_page:
+                break
+        except oci.exceptions.ServiceError as e:
+            if e.code == 'NotAuthorizedOrNotFound':
+                print 'Error: Please check if you have sufficient \
+                permissions to execute the operation' + str(operation.im_func)
+            elif e.status == '500':
+                print 'InternalServerError: Please retry after sometime.'
+            else:
+                print 'Something went wrong.'
+                print 'Here is the info about the exception:'
+                print str(e)
+            quit()
+
+
+def get_compartments():
+    ''' Retrieve the list of compartments under the tenancy
+    '''
+    listofcompartments = []
+    for list_compartments_json in paginate(
+            identity.list_compartments,
+            compartment_id=tenancy_id):
+            listofcompartments.append(str(list_compartments_json.id))
+    return listofcompartments
+
+
+def get_audit_events(start_time, end_time):
+    '''Get events iteratively for each compartment defined in 'compartments' array
+    for the region defined in "config['region']".
+    '''
+    listofauditevents = []
+    for i in range(len(compartments)):
+        for events in paginate(
+                audit.list_events,
+                compartment_id=compartments[i],
+                start_time=start_time,
+                end_time=end_time):
+            '''The following condition eliminates all the read API calls.
+              Uncomment this if write events need to be filtered.
+              '''
+            # if events.request_action != 'GET':
+            listofauditevents.append(events)
+    return listofauditevents
+
+
+'''Here's the main code that retrives audits across the tenancy'''
+
+#  Retrieve the list of all the OCI regions.
+#  This gracefully handles addition of new regions.
+regions = get_regions()
+
+#  Retrieve all the compartments.
+#  This gracefully handles compartment changes on a tenancy.
+compartments = get_compartments()
+
+#  For a region defined by config['region'] get the logs for each compartment
+for i in range(len(regions)):
+    config['region'] = regions[i]
+
+    #  Reintialize with a different region value
+    audit = oci.audit.audit_client.AuditClient(config)
```

Comment:
This re-doing of the configuration and then creating new clients per region is not necessary. Since you're not running this across multiple threads, you can just change the client in-place by calling `set_region()` on it